File: 32-从上到下打印二叉树/main.py
from queue import Queue


# -*- coding:utf-8 -*-
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None
class Solution:
    def PrintFromTopToBottom(self, root):
        # write code here
        if not root:
            return []
        res = []
        q = Queue()
        q.put(root)
        while not q.empty():
            mask = q.get()
            res.append(mask.val)
            if mask.left:
                q.put(mask.left)
            if mask.right:
                q.put(mask.right)
        return res


root = TreeNode(1)
res = Solution().PrintFromTopToBottom(root)
print(res)

File: 32-从上到下打印二叉树/main.cpp
#include <queue>
#include <vector>
using namespace std;
/*
struct TreeNode {
        int val;
        struct TreeNode *left;
        struct TreeNode *right;
        TreeNode(int x) :
                        val(x), left(NULL), right(NULL) {
        }
};*/
class Solution {
public:
  vector<int> PrintFromTopToBottom(TreeNode *root) {
    if (!root) {
      return vector<int>{};
    }
    vector<int> res;
    queue<TreeNode *> q;
    q.push(root);
    while (q.size()) {
      TreeNode *mask = q.front();
      res.push_back(mask->val);
      q.pop();
      if (mask->left) {
        q.push(mask->left);
      }
      if (mask->right) {
        q.push(mask->right);
      }
    }
    return res;
  }
};
